docs/websites/varnish/use-varnish-with-nginx-to-serve-cached-wordpress-content-over-ssl-on-debian-8.md

```diff
@@ -0,0 +1,683 @@
+
+	{: .file-excerpt }
+	/etc/varnish/custom.vcl
+	:   ~~~ conf
+	
+		sub vcl_deliver {
+		if (req.http.X-Purger) {
+		set resp.http.X-Purger = req.http.X-Purger;
+		  }
+		}
+		~~~
+
+	This concludes the `custom.vcl` configuration. You can save and exit the file. The final `custom.vcl` file will look like this:
+
+	{: .file }
+	/etc/varnish/custom.vcl
+	:   ~~~ conf
+	
+		vcl 4.0;
+
+		backend default {
+		.host = "localhost";
+		.port = "8080";
+		}
+
+		acl purger {
+		"localhost";
+		"your_server_ip";    
+		"your_server_ipv6";
+		}
+
+		sub vcl_recv {
+
+		 if (client.ip != "127.0.0.1" && req.http.host ~ "example-over-https.com") {
+			set req.http.x-redir = "https://www.example-over-https.com" + req.url;
+			return(synth(850, ""));
+		 }
+
+		 if (req.method == "PURGE") {
+			if (!client.ip ~ purger) {
+			return(synth(405, "This IP is not allowed to send PURGE requests."));
+		  }
+		 return (purge);
+		 }
+
+		 if (req.restarts == 0) {
+			if (req.http.X-Forwarded-For) {
+			set req.http.X-Forwarded-For = client.ip;
+			}
+		 }
+
+		 if (req.http.Authorization || req.method == "POST") {
+			return (pass);
+		 }
+
+		 if (req.url ~ "/feed") {
+			return (pass);
+		 }
+
+		 if (req.url ~ "wp-admin|wp-login") {
+			return (pass);
+		 }
+
+		 set req.http.cookie = regsuball(req.http.cookie, "wp-settings-\d+=[^;]+(; )?", "");
+		 set req.http.cookie = regsuball(req.http.cookie, "wp-settings-time-\d+=[^;]+(; )?", "");
+
+		 if (req.http.cookie == "") {
+		 unset req.http.cookie;
+		 }
+		}
+
+		sub vcl_synth {
+		 if (resp.status == 850) {  
+			 set resp.http.Location = req.http.x-redir;       
+			 set resp.status = 302;
+			 return (deliver);
+		 }
+		}
+
+		sub vcl_purge {
+		 set req.method = "GET";
+		 set req.http.X-Purger = "Purged";
+		 return (restart);
+		}
+
+		sub vcl_backend_response {
+		 set beresp.ttl = 24h;
+		 set beresp.grace = 1h;
+
+		 if (bereq.url !~ "wp-admin|wp-login|product|cart|checkout|my-account|/?remove_item=") {
+		 unset beresp.http.set-cookie;    
+		 }
+		}
+
+		sub vcl_deliver {
+		 if (req.http.X-Purger) {
+			set resp.http.X-Purger = req.http.X-Purger;
+		 }
+		}
+		~~~
+
+
+	For Varnish to work we also need to edit the `varnish.service` file to use our custom configuration file, with the port number and allocated memory values updated to match the changes we made in our `/etc/default/varnish` file.
+
+21. Open the `/lib/systemd/system/varnish.service` file:
+
+        sudo nano /lib/systemd/system/varnish.service
+
+22. Find the two lines beginning with  `ExecStart` and edit them to look like this:
+
+	{: .file-excerpt }
+	lib/systemd/system/varnish.service
+	:   ~~~ conf
+	
+		ExecStartPre=/usr/sbin/varnishd -C -f /etc/varnish/custom.vcl
+		ExecStart=/usr/sbin/varnishd -a :80 -T localhost:6082 -f /etc/varnish/custom.vcl -S /etc/varnish/secret -s malloc,500M
+		~~~
+
+23. After saving and exiting the file you have to reload the systemd process:
+
+        sudo systemctl daemon-reload
```

Comment:
After running this, Varnish remains running with the previous configuration:

```
alex@afornuto-1:~$ sudo systemctl daemon-reload
alex@afornuto-1:~$ sudo systemctl status varnish
● varnish.service - Varnish HTTP accelerator
   Loaded: loaded (/lib/systemd/system/varnish.service; enabled)
   Active: active (running) since Mon 2016-11-07 16:53:46 UTC; 1h 52min ago
 Main PID: 11606 (varnishd)
   CGroup: /system.slice/varnish.service
           ├─11606 /usr/sbin/varnishd -a :6081 -T localhost:6082 -f /etc/varnish/default....
           └─11626 /usr/sbin/varnishd -a :6081 -T localhost:6082 -f /etc/varnish/default....
```

Of course, trying to restart the service at this point will fail, since nginx is already bound by default to port `80`. So I'm just wondering what's the purpose of this?